app/models/miq_provision_virt_workflow.rb

```diff
@@ -336,23 +343,30 @@ def allowed_templates(options = {})
       end
     end
 
-    allowed_templates_list = source_vm_rbac_filter(vms, condition).to_a
+    # Only select the colums we need
+    vms = vms.select(:id, :name, :guid, :uid_ems, :ems_id, :cloud_tenant_id)
+
+    sql = source_vm_rbac_filter(vms, condition, VM_OR_TEMPLATE_EXTRA_COLS).to_sql
+    allowed_templates_list = ActiveRecord::Base.connection.select_all(sql).to_a
```

Comment:
>Guessing the thought here is because we're just going to convert it back into a hash in `create_hash_struct_from_vm_or_template`.
>
> Is the skipping of the serialization most of the savings here?

I would assume so?  It has been a few months, so I am pretty rusty on this code.

That said, there is literally a gem that does this:

https://github.com/sportngin/find_as_hashes

And in it there are some base metrics there.  But yes, basically since the end result is a hash this code avoids needing to convert to an `ActiveRecord` object and back again.